### ISSUE TITLE
overlapped buffering not being used

### ISSUE THREAD

**Issue by leee** [russinnes/fsync-mdc1200-decode#5]
While I was adding support for file input, I was thinking about how I'd handle reading since I figured I could benefit from being able to seek over a static file, something you can't do with streams.

While quickly examining the existing stream handling, it looks like you're just reading in data as received and not really doing any sort of prebuffering and running the decoders on segments of that.

Were you intending on implementing this in the future, or is this an artifact from it's commit 3 years ago and can be removed?

https://github.com/russinnes/fsync-mdc1200-decode/blob/daff2c38cc54243a5b1f26149b34987426c64456/demod.c#L178-L179

**Comment by russinnes** [comment]
Yes, it was part of a previous version years ago which handle both decoders differently. There are a few stragglers I have not cleaned out yet, it needs a bit of garbage collection.

**Comment by leee** [comment]
Brilliant - if you have no objections, I'd be glad to do a bit of taking out the trash I see.

**Comment by russinnes** [comment]
You bet, I'll keep an eye on it now and merge as needed.
